src/libs/actions/IOU.ts

```diff
@@ -5809,6 +5833,104 @@ function setShownHoldUseExplanation() {
     Onyx.set(ONYXKEYS.NVP_HOLD_USE_EXPLAINED, true);
 }
 
+/**
+ * Sets the `splitShares` map that holds individual shares of a split bill
+ */
+function setSplitShares(transaction: OnyxEntry<OnyxTypes.Transaction>, amount: number, currency: string, newAccountIDs: number[]) {
+    if (!transaction) {
+        return;
+    }
+    const oldAccountIDs = Object.keys(transaction.splitShares ?? {}).map((key) => Number(key));
+    const participantAccountIDsWithoutCurrentUser = newAccountIDs.filter((accountID) => accountID !== userAccountID);
+
+    // Create an array containing unique IDs of the current transaction participants and the new ones
+    const accountIDs = [...new Set<number>([userAccountID, ...participantAccountIDsWithoutCurrentUser, ...oldAccountIDs])];
```

Comment:
NAB - can't we just do `[...newAccountIDs, ...oldAccountIDs]` instead of adding back in the userAccountID we just removed?

Reply:
Thing is, if this is called from the participants selector step, we won't have current user account ID in newAccountIDs, if it's called from a split created from a group chat, it will have it included :/

Reply:
So to be safe we just make sure to always remove it and add it back. Curious if you can see how to make this better!

Reply:
ohh i see. Hm, yeah then this makes sense then

Reply:
Will add a comment

Reply:
I just thought of an improvement to the code too

Reply:
Updated!